src/libraries/Common/src/System/Security/Cryptography/MLDsa.cs

```diff
@@ -68,83 +73,17 @@ protected void ThrowIfDisposed()
-        /// </exception>
-        public static int GetSignatureSizeInBytes(MLDsaAlgorithm algorithm) =>
-            ParameterSetInfo.GetParameterSetInfo(algorithm).SignatureSizeInBytes;
-
-        /// <summary>
-        ///   Gets the size of the ML-DSA secret key for the specified algorithm.
-        /// </summary>
-        /// <param name="algorithm">
-        ///   The specific ML-DSA algorithm to query.
-        /// </param>
-        /// <returns>
-        ///   The size, in bytes, of the ML-DSA secret key for the specified algorithm.
-        /// </returns>
-        /// <exception cref="CryptographicException">
-        ///   <paramref name="algorithm"/> is not a valid ML-DSA algorithm identifier.
-        /// </exception>
-        public static int GetSecretKeySizeInBytes(MLDsaAlgorithm algorithm) =>
-            ParameterSetInfo.GetParameterSetInfo(algorithm).SecretKeySizeInBytes;
-
-        /// <summary>
-        ///   Gets the size of the ML-DSA public key for the specified algorithm.
-        /// </summary>
-        /// <param name="algorithm">
-        ///   The specific ML-DSA algorithm to query.
-        /// </param>
-        /// <returns>
-        ///   The size, in bytes, of the ML-DSA public key for the specified algorithm.
-        /// </returns>
-        /// <exception cref="CryptographicException">
-        ///   <paramref name="algorithm"/> is not a valid ML-DSA algorithm identifier.
-        /// </exception>
-        public static int GetPublicKeySizeInBytes(MLDsaAlgorithm algorithm) =>
-            ParameterSetInfo.GetParameterSetInfo(algorithm).PublicKeySizeInBytes;
-
-        /// <summary>
-        ///   Gets the size, in bytes, of the signature for the current instance.
-        /// </summary>
-        /// <value>
-        ///   The size, in bytes, of the signature for the current instance.
-        /// </value>
-        public int SignatureSizeInBytes => _parameterSetInfo.SignatureSizeInBytes;
-
-        /// <summary>
-        ///   Gets the size, in bytes, of the ML-DSA secret key for the current instance.
-        /// </summary>
-        /// <value>
-        ///   The size, in bytes, of the ML-DSA secret key for the current instance.
-        /// </value>
-        public int SecretKeySizeInBytes => _parameterSetInfo.SecretKeySizeInBytes;
-
-        /// <summary>
-        ///   Gets the size, in bytes, of the ML-DSA public key for the current instance.
-        /// </summary>
-        /// <value>
-        ///   The size, in bytes, of the ML-DSA public key for the current instance.
-        /// </value>
-        public int PublicKeySizeInBytes => _parameterSetInfo.PublicKeySizeInBytes;
-
         /// <summary>
         ///  Releases all resources used by the <see cref="MLDsa"/> class.
         /// </summary>
         public void Dispose()
         {
-            Dispose(true);
-            _disposed = true;
-            GC.SuppressFinalize(this);
+            if (_disposed)
```

Comment:
This now only calls Dispose(true) if it was already disposed...

I assume you meant `if (!_disposed)`?  That's a fine guard, and I guess saves on doing `_key?.Dispose()` in the derived type.

Reply:
my bad - good catch🤦